modules/service/src/main/scala/lucuma/odb/graphql/util/PrettyPrinter.scala

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.odb.graphql.util
+
+import edu.gemini.grackle.Cursor.Env
+import edu.gemini.grackle.Cursor.Env.EmptyEnv
+import edu.gemini.grackle.Cursor.Env.NonEmptyEnv
+import edu.gemini.grackle.PathTerm
+import edu.gemini.grackle.Predicate
+import edu.gemini.grackle.Predicate.ToLowerCase.apply
+import edu.gemini.grackle.Query
+import edu.gemini.grackle.Query.*
+import edu.gemini.grackle.Term
+import org.typelevel.paiges.Doc
+import org.typelevel.paiges.Doc.char
+import org.typelevel.paiges.Doc.intercalate
+import org.typelevel.paiges.Doc.line
+import org.typelevel.paiges.Doc.space
+import org.typelevel.paiges.Doc.str
+import org.typelevel.paiges.Doc.text
+
+/** Paiges combinators for Grackle queries. */
+object PrettyPrinter {
```

Comment:
This has turned out to be helpful for debugging. We can upstream it once we're happy with it.

Reply:
Wow, nice.